File: 0028-find-the-index-of-the-first-occurrence-in-a-string/solution.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if not needle:
            return 0

        if len(needle) > len(haystack):
            return -1

        for i in range(len(haystack) - len(needle) + 1):
            j = 0
            while j < len(needle) and haystack[i + j] == needle[j]:
                j += 1

            if j == len(needle):
                return i

        return -1

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/solution.c
int strStr(char* haystack, char* needle) {
    int haystackLen = strlen(haystack);
    int needleLen = strlen(needle);

    if (needleLen == 0) {
        return 0; // If needle is an empty string, return 0
    }

    int i, j;
    for (i = 0; i <= haystackLen - needleLen; i++) {
        for (j = 0; j < needleLen; j++) {
            if (haystack[i + j] != needle[j]) {
                break;
            }
        }
        if (j == needleLen) {
            return i; // Found the needle in haystack at index i
        }
    }

    return -1; // Needle not found in haystack
}
